script/validate-data/index.ts

```diff
@@ -58,7 +59,10 @@ async function checkWorkflows(folders: string[], allowed_categories: object[]):
 
         const workflowWithErrors = await checkWorkflow(workflowFilePath, propertiesFilePath, allowed_categories);
         if(workflowWithErrors.name && workflow_template_names.size == workflow_template_names.add(workflowWithErrors.name).size) {
-          workflowWithErrors.errors.push(`Workflow template name "${workflowWithErrors.name}" already exists`) 
+          workflowWithErrors.errors.push(`Workflow template name "${workflowWithErrors.name}" already exists`)
+        }
+        if(workflow_template_file_names.size == workflow_template_file_names.add(e.name).size) {
+          workflowWithErrors.errors.push(`Workflow template with filename "${e.name}" already exists`) 
```

Comment:
Good point. We would want them to be considered same files

Reply:
But if yaml has same name then properties file will also have same name. Then it will throw a same template name error as there will be two yaml and one properties file for same template

Reply:
So people can't create file with `yaml` and `yml` and same name

Reply:
Is that true if the properties file is in two different directories?

Reply:
In the above scenario they will be considered different templates. But considering the PR raised on dotcom: https://github.com/github/github/pull/238443
Should we remove the filename check as that would not be needed? As we can't have same filename name in one directory as they would have same properties file and with same filename in different directories should be allowed as that would be able to support in dotcom.

Reply:
I am okay with not having file name check as long as they are handled correctly in dotcom.

Reply:
Yes, if we fix the filename in dotcom, we can remove this fileName check because it doesn't have any significance only template name should be unique.

Reply:
Removed the filename check